J1708 (#19)
hostmobility/documentation: docs/interfaces/j1708.md

---
title: J1708
tags:
  - J1708
  - C61
  - MX-V
  - CT
  - CT GL
---

## Overview

> Some of our platforms has J1708 and operating trough uart or spi communication to a co-cpu. supported baudrate is 9600 (faster might be possible but not tested).

Reviewer: ```suggestion
Some of our platforms feature J1708 operating over UART or SPI via a coprocessor. The supported baud rate is 9600 (faster might be possible has not yet been tested).
```

> To be able to use the j1708 you will need to include /usr/lib/libj1708.so to your application.

Reviewer: ```suggestion
To be able to use the J1708 interface you will need to include `/usr/lib/libj1708.so` in your application.
```

> For for quick test you can use `j1708-test`.

Reviewer: ```suggestion
For for a quick test you can use `j1708-test`.
```

```

> root@mx4-ct-29009999:~# j1708-test

Reviewer: Is this application in $PATH? If not, please provide the absolute path.

Author: yes it is for those unit that has j1708

j1708-test
Usage: "j1708-test" l:n:p:b:m:csrqwBNR [j1708 frame]

Options:
-p,             Priority for sending (default =  8)
-m,             MID (default = 128)
-b,             Baud-rate ([9600], 19200, 115200)
-B,             Bad checksum send
-s,             Send mode. Only one mode can be active
-r,             Receive mode. Only one mode can be active
-R,             Synchronous receive mode. Only one mode can be active
-N,             Non blocking read. Mean to be used together with -R
-l,             Number of loops in send mode.
-q,             Send sequenced data
-w,             Raw data mode, user will get messages with bad CRC
-c,             Crazy send mode. A more sporadic send then -s cmd
-n,             Number of frames to receive before terminating app

Examples:

Send 10 frames with the data provided:
j1708 -s -l 10 1 2 3 4 5 6 7 8 9 10
Read
j1708 -r
```

### MX-V

> This product use spi to communicate to j1708 bus trough a co-cpu.
> 
> The built lib is from our [hm-commercial](https://github.com/hostmobility/hm-commercial)

Reviewer: ```suggestion
This platform uses SPI to communicate with the J1708 bus trough a coprocessor.

The compiled library is from the [hm-commercial repository](https://github.com/hostmobility/hm-commercial).
```

### CT and CT GL

> This product use uart to communicate to j1708 bus trough a co-cpu. The built lib is from our [mx4-commercial](https://github.com/hostmobility/mx4-commercial) or [mx4](https://github.com/hostmobility/mx4) repo

Reviewer: ```suggestion
This platform uses UART to communicate with the J1708 bus via a coprocessor. The compiled library is from our [mx4-commercial](https://github.com/hostmobility/mx4-commercial) or [mx4](https://github.com/hostmobility/mx4) repositories.
```

### C61

> No current product has populated this alternative but it is possible, and J1708 will work the same as CT and CT GL.

Reviewer: ```suggestion
No current product has populated this alternative but it could be done, and J1708 should work the same as for CT and CT GL.
```
